src/scripts/main.js

```diff
@@ -1 +1,69 @@
 'use strict';
+
+document.oncontextmenu = () => false;
+
+const buttonState = { left: false, right: false };
+let resetTimeout;
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1) {
+      return resolve('First');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1 || e.buttons === 2) {
+      return resolve('Second');
+    }
+  });
+});
+
+const promise3 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      buttonState.left = true;
+    } else if (e.button === 2) {
+      buttonState.right = true;
+    }
+
+    if (buttonState.left && buttonState.right) {
```

Comment:
In promise3, the condition 'buttonState.left && buttonState.right' checks if both buttons are pressed. Ensure this is the intended behavior, as it requires both buttons to be pressed simultaneously to resolve the promise.